cypress/integration/shared.js

```diff
@@ -192,6 +198,112 @@ export const waitForClusterInstallation = () => {
+  cy.get('#password').should('have.value', password);
+  cy.get('#kc-form-login').submit();
+
+  // visit ocm environment specified by CYPRESS_HOST_ENV
+  switch (HOSTED_ENV) {
+    case 'staging':
+      cy.visit('');
+      break;
+    case 'integration':
+      cy.visit('', {
+        qs: {
+          env: 'integration',
+        },
+      });
+      break;
+    case 'production':
+      break;
+  }
+};
+
+export const makeApiCall = (apiPostfix, method, responseHandler, requestBody = {}) => {
+  let apiBaseURL = '';
+  // determine ocm api to request user pull secret for comparison
+  switch (HOSTED_ENV) {
+    case 'staging':
+      apiBaseURL = STAGING_API_BASE_URL;
+      break;
+    case 'integration':
+      apiBaseURL = INTEGRATION_API_BASE_URL;
+      break;
+    case 'production':
+      apiBaseURL = PRODUCTION_API_BASE_URL;
+      break;
+  }
+
+  // get ocm api token from cookies
+  cy.getCookie('cs_jwt').then((cookie) => {
+    const token = cookie.value;
+    const authHeader = {
+      Authorization: `Bearer ${token}`,
+    };
+    const requestOptions = {
+      method: method,
+      url: `${apiBaseURL}${apiPostfix}`,
+      headers: authHeader,
+      body: requestBody,
+    };
+
+    // request user pull secret
+    cy.request(requestOptions).then(responseHandler); 
+  });
+};
+
+// verifies auto-filled pull secret matches loged in user's pull secret
+export const verifyPullSecret = () => {
+  // response handler for makeApiCall
+  const comparePullSecret = (response) => {
+    const userPullSecret = JSON.stringify(response.body);
+    cy.get('#form-input-pullSecret-field').should('have.value', userPullSecret);
+  };
+
+  makeApiCall('api/accounts_mgmt/v1/access_token', 'post', comparePullSecret);
+};
+
+export const createClusterHosted = (clusterName) => {
+  // navigate from ocm portal to assisted installer
+  cy.get('button').contains('Create cluster').click();
+  cy.get('[href="/openshift/install"').click();
+  cy.get('[href="/openshift/install/metal"').click();
+  cy.get('[data-testid="ai-button"]').click();
+  verifyPullSecret();
+  cy.get('#form-input-name-field').clear();
+  cy.get('#form-input-name-field').type(clusterName);
+  cy.get('form').submit();
+  cy.get('#button-download-discovery-iso').should('be.visible');
+  cy.get('#form-input-name-field').should('have.value', clusterName);
+};
+
+export const logOutOCM = () => {
+  cy.get('#UserMenu').as('userMenu');
+  cy.get('@userMenu').click();
+  cy.contains('Log out').click();
+};
+
+// verifies cluster was created and associated to user
+export const verifyClusterCreation = (clusterName) => {
+  // response handler for makeApiCall
+  const findClusterInList = (response) => {
+    const clusters = response.body;
+    const checkClusterName = (cluster) => clusterName.localeCompare(cluster.name) === 0;
+
+    cy.expect(clusters.some(checkClusterName)).to.be.true;
+  };
+
+  makeApiCall('api/assisted-install/v1/clusters', 'get', findClusterInList);
```

Comment:
Try to use cy.request(). This is how I traverse the list of clusters from the API and find if mine is created:
    cy.request(API_ENDPOINT + '/api/assisted-install/v1/clusters').then((response) => {
      let i = 0;
      for (i = 0; i < response.body.length; i++) {
        if (response.body[i].name == 'test-cluster-1') {
          break;
        }
      }
      expect(i).to.be.lt(response.body.length);
    });

Reply:
makeApiCall uses cy.request. On the hosted service requests need to be accompanied by a token; makeApiCall gets the token and attaches it to the request